client/lib/domains/email-forwarding/index.js

```diff
@@ -7,4 +13,29 @@ function emailForwardingPlanLimit( plan ) {
 	return ( isBusiness( plan ) ? 100 : 5 );
 }
 
-module.exports = { emailForwardingPlanLimit };
+function validateAllFields( fieldValues ) {
+	return mapValues( fieldValues, ( value, fieldName ) => {
+		const isValid = validateField( {
+			value,
+			name: fieldName
+		} );
+
+		return isValid ? [] : [ 'Invalid' ];
+	} );
+}
+
+function validateField( { name, value } ) {
+	switch ( name ) {
+		case 'mailbox':
+			return value.length > 0 && /^([a-z0-9\._\+\-])*$/i.test( value ) && ! /(^\.)|(\.{2,})|(\.$)/.test( value );
```

Comment:
No need to escape "special" characters [inside character sets](https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Regular_Expressions):
> Character set. This pattern type matches any one of the characters in the brackets, including escape sequences. Special characters like the dot(.) and asterisk (*) are not special inside a character set, so they don't need to be escaped. You can specify a range of characters by using a hyphen, as the following examples illustrate.

Even a hyphen is OK, as long as it's the last one :) So `[a-z0-9._+-]` is fine and more readable IMHO.